core/src/test/java/com/backblaze/b2/client/B2LargeFileStorerTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2019, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.client;
+
+import com.backblaze.b2.client.contentSources.B2ContentSource;
+import com.backblaze.b2.client.exceptions.B2Exception;
+import com.backblaze.b2.client.exceptions.B2InternalErrorException;
+import com.backblaze.b2.client.structures.B2AccountAuthorization;
+import com.backblaze.b2.client.structures.B2FileVersion;
+import com.backblaze.b2.client.structures.B2FinishLargeFileRequest;
+import com.backblaze.b2.client.structures.B2Part;
+import com.backblaze.b2.client.structures.B2UploadPartUrlResponse;
+import org.junit.After;
+import org.junit.Test;
+import org.mockito.Matchers;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Supplier;
+
+import static com.backblaze.b2.client.B2TestHelpers.fileId;
+import static com.backblaze.b2.client.B2TestHelpers.makeSha1;
+import static com.backblaze.b2.client.B2TestHelpers.makeVersion;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/*
+ * Copyright 2019, Backblaze, Inc. All rights reserved.
+ */
+public class B2LargeFileStorerTest {
+
+    private static final long FIVE_MEGABYTES = 5000000;
+
+    private static final long FILE_SIZE = 16000000;
+    private static final long PART_COUNT = 3;
+    private static final long PART_SIZE_FOR_FIRST_TWO = FILE_SIZE / PART_COUNT;
+    private static final long LAST_PART_SIZE = FILE_SIZE - (PART_COUNT - 1) * PART_SIZE_FOR_FIRST_TWO;
+
+    // The file version of the large file we are creating.
+    private final B2FileVersion largeFileVersion = makeVersion(4, 4);
+
+    // Set up the parts of the large file.
+    private final B2Part part1 = new B2Part(fileId(1), 1, PART_SIZE_FOR_FIRST_TWO, makeSha1(1), 1111);
+    private final B2Part part2 = new B2Part(fileId(2), 2, PART_SIZE_FOR_FIRST_TWO, makeSha1(2), 2222);
+    private final B2Part part3 = new B2Part(fileId(3), 3, PART_SIZE_FOR_FIRST_TWO, makeSha1(3), 3333);
```

Comment:
is it funny that part3's size is PART_SIZE_FOR_FIRST_TWO?  :)

Reply:
Good catch! Wasn't using the variable I created above.